.changeset/ninety-ladybugs-itch.md

```diff
@@ -0,0 +1,56 @@
+---
+"@vercel/blob": minor
+"vercel-storage-integration-test-suite": patch
+---
+
+# feat(blob): add manual multipart upload methods
+
+This exposes the three different multipart steps as functions of the SDK. Before this change every multipart upload was uncontrolled, meaning the full data was passed to the SDK and it took care of chunking and uploading.
+Now it's possible to manually upload chunks and start and complete the multipart upload. All of the new functions can be used both on the server and the browser. There are two different API's that can be used.
+
+> When using manual multipart uploads **every upload part has to be of the same size** and **the callside has to take care of memory management and concurrent uploads**!
+
+## Fully manual
+
+Use `createMultipartPut`, `multipartPut` and `completeMultipartPut` to manage the upload.
```

Comment:
Can we rename the individual methods to:
`createMultipartUpload`, `uploadPart`, `completeMultipartUpload` instead? This would more closely map S3 in this case and differentiate from "put"

Reply:
Reasoning: the feature we market is "multipart uploads", not "multipart put" 👍

Reply:
And yes it could be a little confusing with the generic `upload` method of client uploads but that's ok

Reply:
I used the names `createMultipartUpload`, `uploadPart`, `completeMultipartUpload` for the client methods that automatically fetch the client token, just like we have it for `put`. How would you name these?

Reply:
I don't really see the benefit of beeing close to the S3 API, since we want to provide a better DX. I think it doesn't make that much of a difference if we call it `upload` or `put`, it just feels more consistent within our SDK